pkg/ingester/instance.go

```diff
@@ -176,19 +175,33 @@ func (i *instance) getOrCreateStream(pushReqStream logproto.Stream) (*stream, er
 		return nil, httpgrpc.Errorf(http.StatusTooManyRequests, validation.StreamLimitErrorMsg())
 	}
 
+	labels, err := util.ToClientLabels(pushReqStream.Labels)
+	if err != nil {
+		return nil, httpgrpc.Errorf(http.StatusBadRequest, err.Error())
+	}
+	fp := i.getHashForLabels(labels)
 	sortedLabels := i.index.Add(labels, fp)
 	stream = newStream(i.cfg, fp, sortedLabels, i.factory)
-	i.streams[fp] = stream
+	i.streams[pushReqStream.Labels] = stream
+	i.streamsByFP[fp] = stream
+
 	memoryStreams.WithLabelValues(i.instanceID).Inc()
 	i.streamsCreatedTotal.Inc()
 	i.addTailersToNewStream(stream)
 
 	return stream, nil
 }
 
+func (i *instance) getHashForLabels(labels []client.LabelAdapter) model.Fingerprint {
+	var fp uint64
+	lbsModel := client.FromLabelAdaptersToLabels(labels)
+	fp, i.buf = lbsModel.HashForLabels(i.buf, []string(nil)...)
```

Comment:
Should it be `HashWithoutLabels` instead of `HashForLabels` because I feel with empty labels it would not hash any of the labels?